facet-serialize/src/lib.rs

```diff
@@ -473,9 +494,11 @@ where
                                 let fields = peek_struct.fields_for_serialize().count();
```

Comment:
Oh, one really subtle thing that I hate!! this is iterating over all the flattened fields twice: once here to count the number of flattened fields, and once while doing the actual serialization

This is needed so we can call `serializer.start_object(Some(_))`, but I couldn't figure out an efficient way to count the number of flattened fields without just iterating (including allocating a vec)...

If there's a way to impl `ExactSizedIterator` for `FieldsForSerializeIter` then we could avoid this

Reply:
> If there's a way to impl `ExactSizedIterator` for `FieldsForSerializeIter` then we could avoid this

Not sure about this, but maybe we can remove the hint...